modules/wpml/wpml-config.php

```diff
@@ -262,6 +279,83 @@ public function translate_taxonomies( $taxonomies, $hide ) {
 		return $taxonomies;
 	}
 
+	/**
+	 * Translation management for strings in blocks content.
+	 *
+	 * @since 3.3
+	 *
+	 * @param array[] $parsing_rules Rules as Xpath expressions to evaluate in the blocks content.
+	 * @return array[] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>> $parsing_rules
+	 */
+	public function translate_blocks( $parsing_rules ) {
+		return $this->extract_blocks_parsing_rules( $parsing_rules, 'xpath' );
+	}
+
+	/**
+	 * Translation management for block attributes.
+	 *
+	 * @since 3.3
+	 *
+	 * @param array[] $parsing_rules Rules for block attributes to translate.
+	 * @return array[] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>>
+	 */
+	public function translate_blocks_attributes( $parsing_rules ) {
+		return $this->extract_blocks_parsing_rules( $parsing_rules, 'key', true );
+	}
+
+	/**
+	 * Extract what kind of string to translate for blocks from WPML config file.
+	 *
+	 * @since 3.3
+	 *
+	 * @param array[] $parsing_rules         Rules to complete with ones from wpml-config file..
```

Comment:
You can be more specific with `string[][]` according to your PHPStan information.